nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/PublishAMQP.java

```diff
@@ -195,29 +193,85 @@ public void process(final InputStream in) throws IOException {
-                String amqpPropName = attributeEntry.getKey().split("\\" + AMQPUtils.AMQP_PROP_DELIMITER)[1];
+                String amqpPropName = attributeEntry.getKey();
                 String amqpPropValue = attributeEntry.getValue();
-                try {
-                    if (amqpPropertyNames.contains(AMQPUtils.AMQP_PROP_PREFIX + amqpPropName)) {
-                        Method m = builder.getClass().getDeclaredMethod(amqpPropName, String.class);
-                        m.invoke(builder, amqpPropValue);
-                    } else {
-                        getLogger().warn("Unrecogninsed AMQP property '" + amqpPropName + "', will ignore.");
+
+                AMQPUtils.PropertyNames propertyNames = AMQPUtils.PropertyNames.fromValue(amqpPropName);
+
+                if (propertyNames != null) {
+                    try {
+                        switch (propertyNames){
+                            case CONTENT_TYPE:
+                                builder.contentType(amqpPropValue);
+                                break;
+                            case CONTENT_ENCODING:
+                                builder.contentEncoding(amqpPropValue);
+                                break;
+                            case HEADERS:
+                                String[] s = amqpPropValue.split("\\" + AMQPUtils.AMQP_PROP_DELIMITER);
+                                Map<String, Object> headers = new HashMap<>();
+
+                                for (int i = 0; i < s.length ; i += 2){
+                                    if (i + 2 <= s.length){
+                                        headers.put(s[i], s[i + 1]);
+                                    }
+                                }
+
+                                builder.headers(headers);
+                                break;
+                            case DELIVERY_MODE:
+                                builder.deliveryMode(Integer.parseInt(amqpPropValue));
+                                break;
+                            case PRIORITY:
+                                builder.priority(Integer.parseInt(amqpPropValue));
+                                break;
+                            case CORRELATION_ID:
+                                builder.correlationId(amqpPropValue);
+                                break;
+                            case REPLY_TO:
+                                builder.replyTo(amqpPropValue);
+                                break;
+                            case EXPIRATION:
+                                builder.expiration(amqpPropValue);
+                                break;
+                            case MESSAGE_ID:
+                                builder.messageId(amqpPropValue);
+                                break;
+                            case TIMESTAMP:
+                                builder.timestamp(new SimpleDateFormat("yyyy/MM/dd HH:mm:ss").parse(amqpPropValue));
```

Comment:
1 & 2: These errors will be caught by the try/catch block, so if we are going to fully validate each individual one the then we could remove the try/catch? That seems like a lot more code for not-a-lot of benefit?

3: I was under the assumption that the date would be validated before this step and adhere to a certain format anyway (since in my case the programmer was passing the date in, so could adhere to the correct format), but in hindsight that was probably a bit naive, since I guess flows could come from anywhere with any date format that is out of the programmer's control. I will look into the StringDate representation like you mentioned.